### DISCUSSION TITLE
ex-command insertion

### DISCUSSION THREAD

**Discussion by gleachkr** [olimorris/codecompanion.nvim#442] [Ideas]
It would be neat if you could type a desired action on the ex-command line, and have code companion replace it with a functioning nvim command. Something like, `:CodeCompanionEx append the line number to each even numbered line in this file`, and get `:g/^/if line('.') % 2 == 0 | execute "normal! A" . line('.') | endif`.

**Comment by tizot** [comment]
That's exactly what this plugin does: https://github.com/3v0k4/exit.nvim

**Reply by olimorris** [reply]
Thanks for sharing. Okay that's awesome. I'll take a look at exit

**Comment by olimorris** [comment]
I've added `CodeCompanionCmd` in v9.12.0. Let me know your feedback:

https://github.com/user-attachments/assets/c6e5005f-8078-4bb3-8ba6-224072cd4562
